script.js

```diff
@@ -21,27 +21,39 @@ operations.forEach((e, i) => operations[i].addEventListener('click', (e) => oper
 
 let operatorEnter = (op) => {
   let localOperationMemory = display.value;
+  const maxDigits = 10;
+  const decimalPlacesPlusPoint = 6;
   if (memoryNewNumber && memoryPendingOperation !== '=') display.value = memoryCurrentNumber;
-    else { memoryNewNumber = true;
-    if (memoryPendingOperation === '+') memoryCurrentNumber += +localOperationMemory;
-      else if (memoryPendingOperation === '-') memoryCurrentNumber = (memoryCurrentNumber*Math.pow(10, 9) - (+localOperationMemory*Math.pow(10, 9)))/Math.pow(10, 9);
-      else if (memoryPendingOperation === '*') memoryCurrentNumber *= +localOperationMemory;
-      else if (memoryPendingOperation === '/') memoryCurrentNumber /= +localOperationMemory;
-      else memoryCurrentNumber = +localOperationMemory;  
-    (memoryCurrentNumber + '').length < 9 ? display.value = memoryCurrentNumber : display.value = memoryCurrentNumber.toPrecision(6);
+    else { 
+      memoryNewNumber = true;
+    switch (memoryPendingOperation) {
+      case '+':
+        memoryCurrentNumber += +localOperationMemory;
+        break;
+      case '-':
+        memoryCurrentNumber = (memoryCurrentNumber * Math.pow(10, 9) - (+localOperationMemory * Math.pow(10, 9))) / Math.pow(10, 9);
+        break;
+      case '*':
+        memoryCurrentNumber *= +localOperationMemory;
+        break;
+      case '/':
+        memoryCurrentNumber /= +localOperationMemory;
+        break;
+      default:
+        memoryCurrentNumber = +localOperationMemory;
+    }
+    display.value = `${memoryCurrentNumber}`.length < maxDigits ? memoryCurrentNumber : memoryCurrentNumber.toPrecision(decimalPlacesPlusPoint);
     memoryPendingOperation = op;
   } 
 };
 
 clearButtons.forEach((e, i) => clearButtons[i].addEventListener('click', (e) => clearDisplay(e.srcElement.id)));
 
 let clearDisplay = (id) => {
-  if (id === 'ce') {
-    display.value = '0';
-    memoryNewNumber = true;
-  } else if (id === 'c') {
+  const clear = 'c';
```

Comment:
In this case you will declare clear variable each time you enter this function. Such constants should be declared once, let's say, in the top of the script.

Reply:
DONE!